packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -317,7 +317,7 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
     }
 
     private rejectDeferredRealize(reason: string, packageName?: string): never {
-        throw new LoggingError(reason, { packageName: { value: packageName, tag: TelemetryDataTag.PackageData }});
+        throw new GenericFluidError(reason, { packageName: { value: packageName, tag: TelemetryDataTag.PackageData }});
```

Comment:
This will stop these errors from being marked as DataProcessingErrors.  I'm not sure that's desirable though.  Really, I don't understand the various root causes that can lead to this error.  It gets hit all the time, and has for years AFAIK.

@anthony-murphy, @vladsud, @agarwal-navin - Can someone help me think through the best way to classify this error, and what other info is typically needed to diagnose/fix it when it's hit?

Reply:
this method is not very useful. you need to look at the call sites and do the right thing for each

Reply:
maybe you can still have something like this method, or change it to only generate props. in general it just predates a lot of the current logging infra.

Reply:
There are three call sites and they're all in the same function, and seem similar. I'm asking about those cases.

Reply:
@anthony-murphy - Scroll down 20 lines:

![image](https://user-images.githubusercontent.com/12305068/162068593-3abd03d9-b1f7-44a6-ad2b-16b6222270b8.png)

All these cases are flavors of the chain of packages/registries/factories (however you want to describe it) being broken somehow.  I want to know the best way to classify this error by default, and what other info might be useful for diagnosing it.  Said another way - What does the TSG for this error look like?

Reply:
it means the consumer did the wrong things in their container. they either asked for something that wasn't registered, or we tried to load something that should be registered

Reply:
I don't TSG is the right way to think about this. If these are hit there is a code defect in consumer code if we are doing the right thing, or our code if we are doing the wrong thing(unlikely)

Reply:
these come directly from the container and data store authors code. examples:
https://github.com/microsoft/FluidFramework/blob/ce39d612bd9fddb9dedce675e8af77071fd10dea/packages/framework/aqueduct/src/container-runtime-factories/containerRuntimeFactoryWithDefaultDataStore.ts#L27

Reply:
in the load case they should definitely be data processing errors. in the non-load case they should just fail the create/crash the local container

Reply:
i don't know if you have enough context to make that determination at this point in the code

Reply:
in general, i think it is useful to be able to throw an error that some layer above with more context above can annotate, as it can become very unnatural to get all the context in a code path that is called from mutliple entry points like we are here

Reply:
I've made it explicitly DataProcessingError when thrown, rather than depending on the `realize` code path to classify it.

Reply:
> in the load case they should definitely be data processing errors. in the non-load case they should just fail the create/crash the local container

Do you mean container load v. create new?  Anyway, no one who's awaiting `realize()` is logging the error that I can tell - these are only logged by the `RealizeError` event.  🤷‍♂️

Reply:
no i mean create datastore vs load datastore

Reply:
failures in create datastore are local only. nothing hits the wire

Reply:
> Anyway, no one who's awaiting realize() is logging the error that I can tell - these are only logged by the RealizeError event. 🤷‍♂️

Nope, I had a bad query.

Anyhow, resolving this thread.